soroban-sdk/src/bytes.rs

```diff
@@ -102,6 +102,98 @@ macro_rules! bytesn {
     };
 }
 
+#[macro_export]
+macro_rules! impl_bytesn_repr {
+    ($elem: ident, $size: literal) => {
+        impl $elem {
+            pub fn from_bytes(bytes: BytesN<$size>) -> Self {
+                Self(bytes)
+            }
```

Comment:
Could you also add a `from_array` that accepts a `&[u8; $size]`? For symmetry with the `to_array` function being added below.

Reply:
Added the following - 

```
 pub fn from_array(&self, env: &Env, array: &[u8; $size]) -> Self {
      Self(<BytesN<$size>>::from_array(env, array))
 }
```